### ISSUE TITLE
Convert TravisCI to CircleCI

### ISSUE THREAD

**Issue by ctxswitch** [ctxswitch/kafkat#3]
With all the turmoil at Travis these days, it's all the rage to move to Circle to guard against any future disruptions.

**Comment by ctxswitch** [comment]
No matter how much I want to really want to move to CircleCI, I'm putting this one on hold for a little longer.  TravisCI support is still just that much better for ruby and I'd like to focus on coverage and climate - and subsequently the start of the refactor - before messing with testing multiple ruby versions in Circle.  Will come back around on this once the sub command refactor is complete.
